src/conn/conn_api.c

```diff
@@ -1099,6 +1099,12 @@ __conn_close(WT_CONNECTION *wt_conn, const char *config)
     /* Shut down pre-fetching - it should not operate while closing the connection. */
     WT_TRET(__wti_prefetch_destroy(session));
 
+    /*
+     * Shut down background migration. This may perform a checkpoint as part of live restore clean
+     * up, and if it does we need to let the checkpoint complete before continuing.
+     */
+    WT_TRET(__wt_live_restore_server_destroy(session));
```

Comment:
We now call server_destroy earlier in the shutdown process. This is before the `WT_CONN_CLOSING` flag is set.  We still call it in it's original location in `__wti_connection_close`, but it's a no-op on connection close since we've already shut down the server, and is necessary when a connection_open fails and needs to be cleaned up.